server/queue/jobs/deadline.js

```diff
@@ -0,0 +1,61 @@
+const mongoose = require('mongoose');
+const { Board } = require('../../models/Board');
+const connectDB = require('../../db');
+const sendEmail = require('../../utils/sendEmail');
+
+const cardDueTemplate = (dueCards) => {
+  let template = `
+<img src="https://res.cloudinary.com/dpepwhv11/image/upload/v1622628335/logo_hgrobp.png" />
+<div>
+  <h2>Some cards are due today</h2>
+  <p>
+    Hello, just reminding you of the cards that are due today.
+  </p>
+  <div>
+    
+`
+  dueCards.forEach(card => {
+    template += `
+      <div style="background-color: #eee; padding: 20px;">
+        <p>${card.name}</p>
+        <p>Due ${new Intl.DateTimeFormat('en-US').format(new Date(card.dueDate))}</p>
+      </div>
+    `
+  });
+
+  template += '</div>'
+  return template;
+}
+
+/*
+  Deadline reminder job handler.
+*/
+module.exports = async (job, done) => {
+  // spawned processes don't share mongo connection.
+  console.log('establishing mongo connection in spawn...');
+  connectDB();
+  const boards = await Board.find({}).populate('user');
+  if (boards.length === 0) return null;
+
+  const mappedBoard = boards.map((board) => {
+    return {
+      name: board.name,
+      user: board.user,
+      cards: board.columns.map((col) =>
+        col.cards.filter((card) => new Date().getDay() === new Date(card.dueDate).getDay())
+      ).flat()
+    }
+  });
```

Comment:
Unfortunately, the way we designed our back-end this isn't the most efficient way of doing this. If we want to make this more efficient we should consider creating Card models as opposed to having everything built-in.

Reply:
I agree; this could be discussed in a future meeting on how to approach our models issue